### PR TITLE
ControlledThreadPoolExecutorService : Remove threads when they are no longer used

### PR DESCRIPTION
**Pull request by qparis**
At the moment, the threads are not cleaned up when you run muliple script instructions at the same time. (Step to reproduce: run 1+1 multiple time in the console. The number of threads will increase and never decrease) 

Capture d’écran 2019-06-10 à 23 54 33

The allocated memory is therefore never released. This small change allows Phoenicis to close the threads when the pools become empty: 

Capture d’écran 2019-06-10 à 23 52 27

### CODE REVIEW

**Comment by qparis** [comment]
(Codacy can be ignored)

File: .gitpod.dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 FROM gitpod/workspace-full-vnc:latest
 
 RUN dpkg --add-architecture i386
-RUN apt-get update
-RUN apt-get -y install libfreetype6 libfreetype6:i386
+RUN apt-get update && apt-get -y install libxext6 libxext6:i386 libfreetype6 libfreetype6:i386
```
[the remark this answers is not in the document]
Comment (qparis) [reply .gitpod.dockerfile]: This is not but this code is already on master

[the remark this answers is not in the document]
Comment (plata) [reply .gitpod.dockerfile]: Why do we have a diff then? Do you need to merge master to your branch?

[the remark this answers is not in the document]
Comment (qparis) [reply .gitpod.dockerfile]: Yep. I'll do it once this MR is merged

File: phoenicis-multithreading/src/main/java/org/phoenicis/multithreading/ControlledThreadPoolExecutorService.java [hunk 2]
```diff
@@ -48,6 +48,7 @@ public ControlledThreadPoolExecutorService(String name, int numberOfThread, int
     @Override
     public void execute(Runnable runnable) {
         try {
+            this.setCorePoolSize(numberOfThreads);
```
Comment (madoar) [inline phoenicis-multithreading/src/main/java/org/phoenicis/multithreading/ControlledThreadPoolExecutorService.java]: I don't like to call a setter method with a constant during every `execute` invocation this just looks wrong to me.
What about https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html#allowCoreThreadTimeOut(boolean) can we use this alternatively?

Comment (qparis) [reply phoenicis-multithreading/src/main/java/org/phoenicis/multithreading/ControlledThreadPoolExecutorService.java]: You are correct. Will change it in a new PR

**Pull request merged**